staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go

```diff
@@ -765,10 +773,14 @@ func (c *Cloud) updateInstanceSecurityGroupsForNLB(lbName string, instances map[
 
 	{
 		clientPorts := sets.Int64{}
+		clientProtocol := "tcp"
 		healthCheckPorts := sets.Int64{}
 		for _, port := range portMappings {
 			clientPorts.Insert(port.TrafficPort)
 			healthCheckPorts.Insert(port.HealthCheckPort)
```

Comment:
This is going to open the tcp port with the same number as the UDP port for health-checks, as the health-check `updateInstanceSecurityGroupForNLBTraffic` call still specifies TCP.

Reply:
Note that as far as I can tell, when `externalTrafficPolicy: Local` is set, this is the correct behaviour, as all the ports should have the same `HealthCheckPort`, which was reset earlier to be the relevant `healthCheckNodePort`. So this part might be fine, and the issue needs to be resolved elsewhere.

Reply:
Ack.